ledger/src/blockstore_meta.rs

```diff
@@ -269,6 +303,278 @@ impl ShredIndex {
     pub(crate) fn insert(&mut self, index: u64) {
         self.index.insert(index);
     }
+
+    #[allow(unused)]
+    fn remove(&mut self, index: u64) {
+        self.index.remove(&index);
+    }
+}
+
+const NUM_U64_PER_SHRED_INDEX_V2: usize = (MAX_DATA_SHREDS_PER_SLOT + 63) / 64;
+type ShredIndexV2Inner = [u64; NUM_U64_PER_SHRED_INDEX_V2];
```

Comment:
Why using `u64` here? Why not just `u8`?

Reply:
sure, that is the theory.
but I would be surprised if this really moves a needle when running a validator.

For the sake of simplicity these kind of micro-optimizations can be done in follow-up PRs which then can be better tested if they add any value in practice.

Reply:
With all the little endian changes are you still confident that `[u64; N / 64]` instead of `[u8; N / 8]` is not just adding verbosity, complexity and runtime overhead?

Benchmarks do not necessarily capture access patterns on mainnet.
Do you want to run a node on mainnet with `[u8; N / 8]` vs `[u64; N / 64]` to measure if any of the related metrics show an improvement? 

If there is no improvement, then I think this code should just use `[u8; N / 8]` which is simpler (simpler serde, no little endian code, no unsafe, etc) and less verbose. That will save us maintenance churn and it is less likely to introduce bugs either in this PR or if someone changes the code in the future.